crates/astria-composer/src/searcher/mod.rs

```diff
@@ -216,111 +156,96 @@ impl Searcher {
     }
 
     /// Serializes and signs a sequencer tx from a rollup tx.
-    fn handle_pending_tx(&mut self, tx: collector::Transaction) {
+    fn bundle_pending_tx(&mut self, tx: collector::Transaction) {
         let collector::Transaction {
             chain_id,
             inner: rollup_tx,
         } = tx;
 
-        let sequencer_key = self.sequencer_key.clone();
-        let nonce = self.sequencer_nonce.clone();
-
+        // rollup transaction data serialization is a heavy compute task, so it is spawned
+        // on tokio's blocking threadpool
         self.conversion_tasks.spawn_blocking(move || {
-            // Pack into sequencer tx
             let data = rollup_tx.rlp().to_vec();
+            let chain_id = chain_id.into_bytes();
+            let seq_action = Action::Sequence(SequenceAction {
+                chain_id,
+                data,
+            });
 
-            // get current nonce and increment nonce
-            let curr_nonce = nonce.fetch_add(1, Ordering::Relaxed);
-            let unsigned_tx = UnsignedTransaction {
-                nonce: curr_nonce,
-                actions: vec![Action::Sequence(SequenceAction {
-                    chain_id: chain_id.into_bytes(),
-                    data,
-                })],
-            };
-            Ok(unsigned_tx.into_signed(&sequencer_key))
+            vec![seq_action]
```

Comment:
creating a single-item vec seems extreme?

Reply:
the payload of a sequencer transaction is a vector of actions. can make the vector here or in the executor, but doing so here is more correct since the executor should just receive a payload rather than construct it.